cmd/kubelet/app/options/globalflags_linux.go

```diff
@@ -32,48 +29,45 @@ import (
 	_ "github.com/google/cadvisor/machine"
 	_ "github.com/google/cadvisor/manager"
 	_ "github.com/google/cadvisor/storage"
+
+	"k8s.io/component-base/cli/globalflag"
 )
 
 // addCadvisorFlags adds flags from cadvisor
 func addCadvisorFlags(fs *pflag.FlagSet) {
-	// lookup flags in global flag set and re-register the values with our flagset
-	global := flag.CommandLine
-	local := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
```

Comment:
Can you double-check that there's no difference in behavior between registering directly on the target flagset and using fs.AddFlagSet?

Reply:
IMO， no  diff. 
may be have some problem in some case
we replace `local` defined with value `pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)`, using `fs`. but the `fs` defined with value `pflag.NewFlagSet(<component name>, pflag.ContinueOnError)`, so may occurred problem when `os.Args[0]` (the binary name) diff with `<component name>`